test/e2e/clusterctl_upgrade.go

```diff
@@ -127,6 +128,8 @@ type ClusterctlUpgradeSpecInput struct {
 	// If not set, the test will upgrade once to the v1beta1 contract.
 	// For some examples see clusterctl_upgrade_test.go
 	Upgrades []ClusterctlUpgradeSpecInputUpgrade
+
+	SkipMachinePoolMachines bool
```

Comment:
Can we have a godoc comment that this is used for providers which don't support machine pool machines (don't remember the exact wording)

Reply:
I'm not sure how I can get the infrastructure MachinePool in the E2E context without knowing what provider/infrastructure is involved. Is there a way to get the infrastructure MachinePool involved with just the CAPI MachinePool? 

For example, if I want to get an unstructure list of the infra machinepools, I'll need to know the kind of the infrastructure: https://github.com/willie-yao/cluster-api/blob/mp-upgrade/test/e2e/clusterctl_upgrade.go#L444